Feature/create postcode search builder

nshumoogum opened the pull request:
Create the postcode search builder as a separate azure function.

Check unit tests pass and all documentation makes sense.

nshumoogum committed:
Add MIT license

nshumoogum committed:
Add contributing guidelines

nshumoogum committed:
Add github pull request template

nshumoogum committed:
Add build pipelines for all environments

nshumoogum committed:
Add dependencies for azure function application

nshumoogum committed:
Add gitignore file

nshumoogum committed:
Add gitignore file

Used to ignore files we dont want to commit to github

nshumoogum committed:
Create Postcode search builder azure function

nshumoogum committed:
Update README.md

nshumoogum committed:
Create unit tests for modelling postcode resource

Also uncomment out running tests in ci build .yml files foe environments

philratcliffe commented on PostcodeSearchBuilderBlobTrigger/__init__.py:
```diff
@@ -0,0 +1,62 @@
+"""Contains the entry point for Create Postode Search
```
typo Postcode

philratcliffe commented on PostcodeSearchBuilderBlobTrigger/models.py:
```diff
@@ -0,0 +1,86 @@
+    return invalid
+
+def is_float(value):
+  try:
```
looks like the indentation is not consistent in a couple of places?

philratcliffe commented on PostcodeSearchBuilderBlobTrigger/search.py:
```diff
@@ -0,0 +1,172 @@
+            raise exceptions.StopEtlPipelineErrorException
+
+    def create(self):
+        self.get_index()
```
Not an issue, but looks slightly odd calling a function with the name get_... that doesn't return something?

philratcliffe commented on PostcodeSearchBuilderBlobTrigger/search.py:
```diff
@@ -0,0 +1,172 @@
+
+            raise exceptions.StopEtlPipelineErrorException
+
+    def get_index(self):
```
As this doesn't return anything, would a better name be build_.... e.g., build_postcode_schema?

philratcliffe commented on PostcodeSearchBuilderBlobTrigger/search.py:
```diff
@@ -0,0 +1,172 @@
+            self.postcode_schema = schema
+
+
+class Load():
```
Not an issue, but perhaps a better name for this class be Loader? And postcode_documents could be, say, load_postcode_documents?

philratcliffe commented on PostcodeSearchBuilderBlobTrigger/tests/models_test.py:
```diff
@@ -0,0 +1,125 @@
+import unittest
```
👍 Good tests

philratcliffe commented on README.md:
```diff
@@ -1,6 +1,6 @@
-# PostcodeSearchBuilderBlobTrigger - Python
+PostcodeSearchBuilderBlobTrigger - Python
```
👍 Helpful README

philratcliffe commented on README.md:
```diff
@@ -44,4 +44,80 @@ Add the following to your local.settings.json:
+
+5) Create your local.settings.json file at root level of repository and include all environment variables in the configuration settings table above.
+
+6) Create a virtual machine to run the azure function application by running `venv .env` at root level of repository.
```
Perhaps use the phrase Python virtual env rather than virtual machine?

philratcliffe reviewed (approved):
LGTM just a few comments.

nshumoogum committed:
Fix typos, incorrect indentation and rename classes

1) Fix typo
2) Correct indentation (4 spaces)
3) Rename get_index function to build_postcode_schema
4) Rename class Load to Loader
5) Use Python virtual env in README.md rather than virtual machine,
when running application (azure function)

nshumoogum committed:
Correct syntax and line indentation

Also rename functions from camel case to snake case
